ssl/statem/statem_lib.c

```diff
@@ -1571,7 +1571,7 @@ static int is_tls13_capable(const SSL_CONNECTION *s)
     if (s->psk_find_session_cb != NULL || s->cert->cert_cb != NULL)
         return 1;
 
-    for (i = 0; i < SSL_PKEY_NUM; i++) {
+    for (i = 0; i < s->ssl_pkey_num; i++) {
         /* Skip over certs disallowed for TLSv1.3 */
         switch (i) {
         case SSL_PKEY_DSA_SIGN:
```

Comment:
We assume here that all provider based sigs are allowed in TLSv1.3? IIUC that's a requirement anyway, so possibly its ok - but maybe a comment about this somewhere?

Reply:
So added in 0ac814f8247d1a5f157eec2b62bce0000c3a2d1e. TLS1.3 check also made more strict.